pkg/assembler/backends/neo4j/certifyVuln.go

```diff
@@ -17,11 +17,452 @@ package neo4jBackend
 
 import (
 	"context"
-	"fmt"
+	"strings"
 
 	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j/dbtype"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+const (
+	dbUri          string = "dbUri"
+	dbVersion      string = "dbVersion"
+	scannerUri     string = "scannerUri"
+	scannerVersion string = "scannerVersion"
 )
 
 func (c *neo4jClient) CertifyVuln(ctx context.Context, certifyVulnSpec *model.CertifyVulnSpec) ([]*model.CertifyVuln, error) {
-	panic(fmt.Errorf("not implemented: CertifyVuln - CertifyVuln"))
+	err := checkCertifyVulnInputs(certifyVulnSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	queryAll := false
+	if certifyVulnSpec.Osv == nil && certifyVulnSpec.Cve == nil && certifyVulnSpec.Ghsa == nil {
+		queryAll = true
+	}
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
+	aggregateCertifyVuln := []*model.CertifyVuln{}
+
+	if queryAll || certifyVulnSpec.Cve != nil {
+
+		var sb strings.Builder
+		var firstMatch bool = true
+		queryValues := map[string]any{}
+
+		// query CVE
+		returnValue := " RETURN type.type, namespace.namespace, name.name, version.version, version.subpath, " +
+			"version.qualifier_list, certifyVuln, cveYear.year, cveID.id"
+
+		// query with pkgVersion
+		query := "MATCH (root:Pkg)-[:PkgHasType]->(type:PkgType)-[:PkgHasNamespace]->(namespace:PkgNamespace)" +
+			"-[:PkgHasName]->(name:PkgName)-[:PkgHasVersion]->(version:PkgVersion)" +
+			"-[:subject]-(certifyVuln:CertifyVuln)-[:is_vuln_to]-(cveID:CveID)<-[:CveHasID]" +
```

Comment:
same comment but also just putting a comment since there's a chain of PRs, so let you close comments in order of merging.